### PR TITLE
Fix mutator listing in struct-auto-info. Closes #3501.

### LINKED ISSUE #3501: define-struct creates struct-auto-info incorrectly
**What version of Racket are you using?**
7.9 [cs]

**What program did you run?**

```
#lang racket 
(require (for-syntax syntax/parse racket/struct-info))
(struct a-struct (a [b #:mutable] [c #:auto] [d #:auto #:mutable]))
(define-syntax (from-id-get-auto-info stx)
  (syntax-parse stx
    [(_ name:id)
     #`'#,(struct-auto-info-lists (syntax-local-value #'name))]))
(from-id-get-auto-info a-struct)
```

**What should have happened?**

It should not have `set-a-struct-b!`.

**If you got an error message, please include it here.**

N/A

**Please include any other relevant details**

Bug discovered by @kurinoku.

### PR DESCRIPTION
**Pull request by kurinoku**
Closes #3501.

### CODE REVIEW

File: racket/collects/racket/private/define-struct.rkt [hunk 1]
```diff
@@ -613,7 +617,8 @@
                                   (cons (field-to-selector/mutator-directive (car fields)
                                                                              this-set
                                                                              #f)
-                                        other-directives))])))
+                                        other-directives)
+                                  count**)])))
```
Comment (sorawee) [inline racket/collects/racket/private/define-struct.rkt:621]: I didn't see `count**` defined. Did I miss anything?

Comment (sorawee) [reply racket/collects/racket/private/define-struct.rkt:621]: Ah, nvm, I was blind.

**Comment by sorawee** [comment]
This looks good to me. Some comments:

- Could you add a test to https://github.com/racket/racket/blob/master/pkgs/racket-test-core/tests/racket/struct.rktl?
- You could also avoid the accumulator `count`. Instead, return `0` in the base case, and in the recursive case, add 1 to the returned value when the condition applies.

File: racket/collects/racket/private/define-struct.rkt (hunk 1 above)
Comment (kurinoku) [reply racket/collects/racket/private/define-struct.rkt:621]: At L607

**Review by samth — COMMENTED** [review]
Looks good with a test.

**Comment by samth** [comment]
Looks good. Can you squash this into one commit?

**Pull request merged**